src/vcpkg/commands.build.cpp

```diff
@@ -708,6 +708,11 @@ namespace vcpkg
                 {
                     compiler_info.id = s.substr(s_id_marker.size()).to_string();
                 }
+                static constexpr StringLiteral s_path_marker = "#COMPILER_CXX_PATH#";
+                if (Strings::starts_with(s, s_path_marker))
+                {
+                    compiler_info.path = s.substr(s_path_marker.size()).to_string();
+                }
```

Comment:
I don't think this works because detect_compiler isn't going to emit this `COMPILER_CXX_PATH` value: https://github.com/microsoft/vcpkg/blob/21fa4ccecfefd96bb16faac4af17fcb900d4f8b3/scripts/detect_compiler/CMakeLists.txt#L18